Sources/Collar/Classes/Collar/AnalyticsCollectionManager.swift

```diff
@@ -8,134 +8,66 @@
 
 import Foundation
 
-public enum LogType: String {
-    case userProperty = "User property"
-    case event = "Event"
-    case screen = "Screen view"
-}
-
-public struct LogItem: CustomStringConvertible, Identifiable {
-    public let id = UUID()
-    public let type: LogType
-    public let name: String
-    public let timestamp: Date
-    public let value: String?
-    public let parameters: [String: Any]?
-
-    static let dateFormatter = ISO8601DateFormatter()
-
-    init(screenName: String, screenClass: String?) {
-        self.init(type: .screen, name: screenName, value: screenClass)
-    }
-    
-    init(event: String, timestamp: Date, parameters: [String: Any]?) {
-        self.init(type: .event, name: event, timestamp: timestamp, parameters: parameters)
-    }
-    
-    init(userProperty: String, value: String?) {
-        self.init(type: .userProperty, name: userProperty, value: value)
-    }
-    
-    init(type: LogType, name: String, timestamp: Date = Date(), value: String? = nil, parameters: [String: Any]? = nil) {
-        self.type = type
-        self.name = name
-        self.timestamp = timestamp
-        self.value = value
-        self.parameters = parameters
-    }
+public final class AnalyticsCollectionManager: @unchecked Sendable {
 
-    public var description: String {
-        var lines: [String] = []
-        lines.append("Type: \(type.rawValue)")
-        lines.append("Name: \(name)")
-        lines.append("Timestamp: \(LogItem.dateFormatter.string(from: timestamp))")
-        if let value = value {
-            switch type {
-            case .screen:
-                lines.append("Screen class: \(value)")
-            default:
-                lines.append("Value: \(value)")
-            }
-        }
-        if let params = paramsJSONString {
-            lines.append("Parameters: \(params)")
+    public enum Notification {
+        public static var didUpdateLogs: Foundation.Notification {
+            Foundation.Notification(name: .init("AnalyticsCollectionManager.didUpdateLogs"))
         }
-        return lines.joined(separator: "\n")
     }
-}
 
-public class AnalyticsCollectionManager {
-    
-    public enum Notification {
-        public static var didUpdateLogs = Foundation.Notification(name: .init("AnalyticsCollectionManager.didUpdateLogs"))
-    }
-    
     public static let shared = AnalyticsCollectionManager()
-    
-    public private(set) var logs: [LogItem] = [] {
-        didSet {
-            NotificationCenter.default.post(AnalyticsCollectionManager.Notification.didUpdateLogs)
-        }
+
+    private let queue = DispatchQueue(label: "analytics.logs.queue", attributes: .concurrent)
+    private var _logs: [LogItem] = []
+
+    public var logs: [LogItem] {
+        queue.sync { _logs }
```

Comment:
I don't really like to use sync. Maybe some documentation is needed here, like don't call this from the main thread or something similar since this will suspend that thread.